pageserver/src/tenant/timeline/eviction_task.rs

```diff
@@ -403,7 +403,28 @@ impl Timeline {
         // Make one of the tenant's timelines draw the short straw and run the calculation.
         // The others wait until the calculation is done so that they take into account the
         // imitated accesses that the winner made.
-        let mut state = tenant.eviction_task_tenant_state.lock().await;
+        let (mut state, _permit) = {
+            if let Ok(locked) = tenant.eviction_task_tenant_state.try_lock() {
+                (locked, permit)
+            } else {
+                // we might need to wait for a long time here in case of pathological synthetic
```

Comment:
I'm trying to figure out when we would take this branch: `eviction_task_tenant_state` is only locked in `imitate_layer_accesses`, and that function is only called from `eviction_iteration` (either via imitiate_only or via eviction_iteration_threshold).

So when we reach this code, who else could be holding the lock?

Reply:
Each **timeline** has an eviction task so one of them.

As @problame noted, this use of background task permits for *per-timeline* eviction task instead of having a separate *per-tenant* task for doing the per-tenant imitation is wrong. Splitting it might be a sensible move after this, but wouldn't we still need these permits for *per-timeline* imitation as well? I think we will want them for that as well.

Imitation itself will remain to be useful even when the eviction part is no longer used.